tensorflow/core/kernels/resize_nearest_neighbor_op.cc

```diff
@@ -152,19 +153,26 @@ struct ResizeNearestNeighbor<CPUDevice, T, half_pixel_centers, align_corners> {
         if (half_pixel_centers) {
           in_y = std::max(static_cast<Eigen::Index>(0), in_y);
         }
-        for (Eigen::Index x = 0; x < out_width; ++x) {
-          Eigen::Index in_x = std::min(
-              (align_corners)
-                  ? static_cast<Eigen::Index>(roundf(scaler(x, width_scale)))
-                  : static_cast<Eigen::Index>(floorf(scaler(x, width_scale))),
-              in_width - 1);
-          if (half_pixel_centers) {
-            in_x = std::max(static_cast<Eigen::Index>(0), in_x);
-          }
-          std::copy_n(&input(b, in_y, in_x, 0), channels, &output(b, y, x, 0));
+        Eigen::Index in_x = std::min(
+            (align_corners)
+                ? static_cast<Eigen::Index>(roundf(scaler(x, width_scale)))
+                : static_cast<Eigen::Index>(floorf(scaler(x, width_scale))),
+            in_width - 1);
+        if (half_pixel_centers) {
+          in_x = std::max(static_cast<Eigen::Index>(0), in_x);
         }
+        std::copy_n(&input(bs, in_y, in_x, 0), channels, &output(bs, y, x, 0));
       }
-    }
+    };
+    Eigen::Index N = batch_size * out_height * out_width;
+    const int input_bytes =
+        batch_size * in_height * in_width * channels * sizeof(T);
+    const int output_bytes = N * channels * sizeof(T);
+    const int compute_cycles = (Eigen::TensorOpCost::ModCost<T>() * 2 +
+                                Eigen::TensorOpCost::DivCost<T>() * 5) *
```

Comment:
I only see 3 divisions. Where are the other 2 divisions? If they are for the implicit division in mod, I think `ModCost` has already accounted for that.

Also, we call `scaler` twice, each takes 1 addition and 1 multiplication. Maybe add them as well?

Reply:
Thanks, Changes as requested.